synphot/wrapper.py

```
@@ -0,0 +1,153 @@
+        self.obs = Observation(self.source_spec, self.throughput, force=force)
+        self.countrate = self.obs.countrate(area=self.aperture_area)
+        self.snr = ccd_snr(self.countrate * self.exptime)
+
+    def _set_bandpass(self):
+
+        if type(self._bandpass) == list:
+            if len(self._bandpass) != 2:
+                raise exceptions.BandpassError("bandpass should be "
+                                               "a list with 2 x n "
+                                               "dimensions.")
+            # order of waves/transmittance given by bandpass doesn't matter:
+            try:
+                self.bandpass = _call_SpecElement_4list(self._bandpass[0],
+                                                        self._bandpass[1])
+            except u.UnitsError:
+                try:
+                    self.bandpass = _call_SpecElement_4list(self._bandpass[1],
+                                                            self._bandpass[0])
+                except u.UnitsError:
+                    print("rows of bandpass list must be of type "
+                          "astropy.units.quantity.Quantity")
+                
+        elif type(self._bandpass) == str:
+            self.bandpass = SpectralElement.from_filter(self._bandpass)
+
+        elif type(self._bandpass) == SpectralElement:
+            self.bandpass = self._bandpass
+
+        else:
+            raise exceptions.BandpassError("bandpass must be of type str, "
+                "`~synphot.spectrum.SpectralElement`, or "
+                "list of shape(2, n) containing "
+                "`~astropy.units.quantity.Quantity` values")
+
+        self.throughput = self.bandpass
+
+    def _include_qe(self):
+        if len(self._qe) != 2:
+            raise exceptions.BandpassError("quantum_efficiency should be "
+                                           "a list with 2 x n "
+                                           "dimensions.")
+        try:
```

Reminder RE: slack convo to let the function fail when the user supplies the wrong inputs, rather than writing the same exception block in a bunch of places. Bumpers are nice but at some point obfuscate your code.

---

Since I've moved these bits to a function (where the exception is only stated once) it's okay to keep?